src/dr_log.cpp

```diff
@@ -229,13 +269,20 @@ void setupLogging(std::string const & log_file, std::string const & name) {
 	core->add_global_attribute("Node", log::attributes::constant<std::string>(name));
 
 	// Get environment variables to check if we should skip some sinks.
-	char * use_console_sink = std::getenv("DR_LOG_USE_CONSOLE");
-	char * use_syslog_sink  = std::getenv("DR_LOG_USE_SYSLOG");
+	char * use_console_sink   = std::getenv("DR_LOG_USE_CONSOLE");
+	char * use_syslog_sink    = std::getenv("DR_LOG_USE_SYSLOG");
+	char * console_format    = std::getenv("DR_LOG_CONSOLE_FORMAT");
+
+	bool console_sink_enabled = ((!use_console_sink || std::atoi(use_console_sink)) && (!console_format || (strcmp("systemd", console_format) != 0)));
+	bool syslog_sink_enabled  = (!use_syslog_sink  || std::atoi(use_syslog_sink));
+	bool systemd_sink_enabled = ((!use_console_sink || std::atoi(use_console_sink)) && (!console_format || (strcmp("systemd", console_format) == 0)));
```

Comment:
This was already broken before, so it's not your fault, but: this is not a good way to process environment variables.

`std::getenv` returns a `nullptr` if the environment variable is not set. Neither `std::atoi` nor `std::strcmp` (should never use plain `strcmp` in C++) are prepared to deal with that.

I would recommend to add a local utility function 'getEnv` that returns an `std::string` and checks for null pointers. It could just return an empty string for non-existing env vars.

Then you can also just compare the string with `==` instead of using `std::strcmp`.

I also wouldn't call `std::atoi/std::stoi`. I would just compare the value with the string `"0"`. Much easier, and actually fully equivalent since `std::atoi/std::stoi` stop parsing when they hit non-digit characters. Even cheaper, although that really isn't important.

Reply:
Sounds good, done!